File: android-compat/src/main/java/android/util/Log.java
package android.util;

public final class Log {
    public static final int VERBOSE = 2;
    public static final int DEBUG = 3;
    public static final int INFO = 4;
    public static final int WARN = 5;
    public static final int ERROR = 6;

    public static int v(String tag, String msg) { System.out.println("V/" + tag + ": " + msg); return 0; }
    public static int v(String tag, String msg, Throwable tr) { System.out.println("V/" + tag + ": " + msg); return 0; }
    public static int d(String tag, String msg) { System.out.println("D/" + tag + ": " + msg); return 0; }
    public static int d(String tag, String msg, Throwable tr) { System.out.println("D/" + tag + ": " + msg); return 0; }
    public static int i(String tag, String msg) { System.out.println("I/" + tag + ": " + msg); return 0; }
    public static int i(String tag, String msg, Throwable tr) { System.out.println("I/" + tag + ": " + msg); return 0; }
    public static int w(String tag, String msg) { System.err.println("W/" + tag + ": " + msg); return 0; }
    public static int w(String tag, String msg, Throwable tr) { System.err.println("W/" + tag + ": " + msg); return 0; }
    public static int w(String tag, Throwable tr) { System.err.println("W/" + tag); return 0; }
    public static int e(String tag, String msg) { System.err.println("E/" + tag + ": " + msg); return 0; }
    public static int e(String tag, String msg, Throwable tr) { System.err.println("E/" + tag + ": " + msg); return 0; }
    public static boolean isLoggable(String tag, int level) { return true; }
    public static int println(int priority, String tag, String msg) { return 0; }
    public static String getStackTraceString(Throwable tr) { return tr.toString(); }
}

File: core/src/commonMain/kotlin/ireader/core/log/Log.kt
//package ireader.core.log
//
//import co.touchlab.kermit.Logger
//import co.touchlab.kermit.Severity
//
///**
// * Logging utility for the core module using Kermit.
// * This is a separate implementation from source-api's Log to avoid circular dependencies.
// */
//object Log {
//
//    private const val DEFAULT_TAG = "IReader"
//
//    /**
//     * Minimum severity level for logging.
//     */
//    var minSeverity: Severity = Severity.Info
//
//    /**
//     * Logs a lazy message at debug level.
//     */
//    fun debug(message: () -> String) {
//        if (minSeverity.ordinal <= Severity.Debug.ordinal) {
//            Logger.d(DEFAULT_TAG) { message() }
//        }
//    }
//
//    /**
//     * Logs a message at debug level with tag.
//     */
//    fun debug(message: String, tag: String = DEFAULT_TAG) {
//        if (minSeverity.ordinal <= Severity.Debug.ordinal) {
//            Logger.d(tag) { message }
//        }
//    }
//
//    /**
//     * Logs a lazy message at info level.
//     */
//    fun info(message: () -> String) {
//        if (minSeverity.ordinal <= Severity.Info.ordinal) {
//            Logger.i(DEFAULT_TAG) { message() }
//        }
//    }
//
//    /**
//     * Logs a message at info level with tag.
//     */
//    fun info(message: String, tag: String = DEFAULT_TAG) {
//        if (minSeverity.ordinal <= Severity.Info.ordinal) {
//            Logger.i(tag) { message }
//        }
//    }
//
//    /**
//     * Logs a lazy message at warn level.
//     */
//    fun warn(message: () -> String) {
//        Logger.w(DEFAULT_TAG) { message() }
//    }
//
//    /**
//     * Logs a message at warn level with tag.
//     */
//    fun warn(message: String, tag: String = DEFAULT_TAG) {
//        Logger.w(tag) { message }
//    }
//
//    /**
//     * Logs a lazy message at error level.
//     */
//    fun error(message: () -> String) {
//        Logger.e(DEFAULT_TAG) { message() }
//    }
//
//    /**
//     * Logs a message at error level with tag.
//     */
//    fun error(message: String, tag: String = DEFAULT_TAG) {
//        Logger.e(tag) { message }
//    }
//
//    /**
//     * Logs an exception at error level with lazy message.
//     */
//    fun error(exception: Throwable, message: () -> String) {
//        Logger.e(DEFAULT_TAG, exception) { message() }
//    }
//
//    /**
//     * Logs an exception at error level.
//     */
//    fun error(exception: Throwable, message: String? = null, tag: String = DEFAULT_TAG) {
//        Logger.e(tag, exception) { message ?: exception.message ?: "Unknown error" }
//    }
//}
